src/Sylius/Component/Core/Promotion/Filter/TaxonFilterInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Promotion\Filter;
+
+use Doctrine\Common\Collections\Collection;
+use Sylius\Component\Core\Model\OrderItemInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+interface TaxonFilterInterface
+{
+    /**
+     * @param Collection $items
```

Comment:
We only need to iterate through them. The bad thing is that `function foo(\Traversable $x)` does not allow to pass an array as an argument and `function foo(array $x)` does not allow to pass an `\Traversable` object.

IMHO the best option would be to use an `array` and `Collection::toArray()` to get an array from collection.